adaptivefiltering/apps.py

```diff
@@ -623,88 +623,128 @@ def assign_pipeline(dataset, segmentation, pipelines):
     """
 
     # passes the segment to the _update_seg_pin function
-    def on_button_clicked(b, layer_data=None):
-        return _update_seg_marker(layer_data)
+    def on_button_clicked(b, segmentation=None):
+        return _update_seg_marker(segmentation)
 
     # holds a segmentation and calculates
-    def _update_seg_marker(layer_data):
+    def _update_seg_marker(segmentation):
         # initilizes a new marker
-        layer_data.properties["style"]["color"] = "red"
-        layer_data.properties["style"]["fillOpacity"] = "0"
+        for feature in segmentation["features"]:
+            feature["properties"]["style"]["color"] = "red"
+            feature["properties"]["style"]["fillOpacity"] = "0"
 
         for layer in map_.map.layers:
             if layer.name == "Current Segmentation":
                 map_.map.remove_layer(layer)
 
-        map_.load_geojson(layer_data, "Current Segmentation")
+        map_.load_geojson(segmentation, "Current Segmentation")
 
     def _create_right_side_menu():
         right_side_label = ipywidgets.Label("Assign Pipelines to Segmentations")
+        from adaptivefiltering.utils import split_segmentation_classes
 
-        # pipeline author has to be replaced with the storage location
+        # needed to quickly check if all features have the same keys
 
-        # the no pipeline option ensures, that the user picks one pipeline.
-        dropdown_options = [("no Pipeline", "")] + [
-            (pipeline.title, pipeline.author) for pipeline in pipelines
+        split_segmentation = split_segmentation_classes(segmentation)
+
+        feature_properties_options = [
+            (keys, keys) for keys in split_segmentation.keys()
```

Comment:
This behaviour can also be achieved by simple passing the list of keys. See https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20List.html#Dropdown